index.js

```diff
@@ -21,12 +21,24 @@ function parse(req) {
       label: query.label || ((query.compression ? 'gzip ' : '') + 'size'),
       color: query.color || 'brightgreen',
       style: query.style || null,
+      max: query.max || query.threshold || null,
+      threshold: query.threshold || query.max || null,
       value: 'unknown',
       extension: 'svg',
       size: 0,
       compression: query.compression,
       compressedSize: 0,
-      err: null
+      err: null,
+      updateColor: function() {
+        if (this.max) {
+          let thresh = [this.max, this.threshold, this.size].sort((a, b) => a - b)
+          switch (this.size) {
+            case (thresh[0]) : this.color = 'brightgreen' ; break
+            case (thresh[1]) : this.color = 'yellow' ; break
+            case (thresh[2]) : this.color = 'red' ; break
+          }
+        }
+      }
     }
```

Comment:
`baton` is just a POJO, and does not contain any function. Can you put this function in the global scope as other operations (ie. `fetch`, `compressed`, ...)?

Also I would simplify the implementation a bit:
```js
function updateColor(baton) {
  if (!baton.max) return baton

  if (baton.size > baton.max) {
    baton.color = 'red'
  }
  else if (baton.threshold && baton.size > baton.threshold) {
    baton.color = 'yellow'
  }

  return baton
}
```